src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1573,6 +1576,430 @@ LedgerManagerImpl::prefetchTransactionData(AbstractLedgerTxnParent& ltx,
     }
 }
 
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+
+ThreadEntryMap
+LedgerManagerImpl::collectEntries(AbstractLedgerTxn& ltx, Thread const& txs)
+{
+    ThreadEntryMap entryMap;
+    auto getEntries = [&](TransactionFrameBasePtr tx,
+                          MutableTxResultPtr resPayload,
+                          xdr::xvector<LedgerKey> const& keys,
+                          uint32_t& readBytes) {
+        for (auto const& lk : keys)
+        {
+            if (!resPayload->isSuccess())
+            {
+                // This tx has already failed, so no need to load anything
+                break;
+            }
+
+            auto ltxe = ltx.loadWithoutRecord(lk);
```

Comment:
Shouldn't we also populate the ledger load metrics here? (currently these are a part of the soroban ops implementation). 
Also, I see the code for these metrics remains in the operations, but in case if the limit is exceeded, we will never reach the operation code at all (so at least the error handling code in the operations seems redundant).

Reply:
Done